user_deprovision/user_deprovision.py

```diff
@@ -148,43 +179,42 @@ def resolve_incidents(self, incidents):
             List of incident-like dict objects, i.e. retrieved from the API
         """
         for incident in incidents:
-            log.info('Resolving %s', incident['id'])
-            try:
-                self.rput(incident['self'],
-                    json={'type':'incident_reference', 'status':'resolved'})
-            except PDClientError as e:
-                handle_exception(e)
-                log.error("Could not resolve incident %s.", incident['id'])
+            if len(incident.get('assignments')) > 1:
```

Comment:
Note: this is the only change in terms of desired functionality of the script. I noticed that we were resolving incidents that had additional responders, so I put this check in place to prevent that.

Reply:
I think we have to resolve open incidents in order to be able to delete the user. Incidents use an escalation policy snapshot so as long as the incident is still associated with that it can't remove the user. I can send you an example from my testing where the user could not be deleted.

Reply:
Ah, interesting. I'll add that to the readme then cause it's something we should be warning customers about.

Reply:
double-take: looks like it can successfully delete users even if they are still on incidents.

Reply:
quadruple take: jade's right. code removed.